diff/parse.go

```diff
@@ -436,12 +435,6 @@ func (r *HunksReader) ReadHunk() (*Hunk, error) {
 			r.hunk.Body = append(r.hunk.Body, '\n')
 		}
 	}
-
-	// Final hunk is complete. But if we never saw a hunk in this call to ReadHunk, then it means we hit EOF.
-	if r.hunk != nil {
-		return r.hunk, nil
-	}
-	return nil, io.EOF
```

Comment:
This looks like code unintentionally left behind after it's no longer needed. It was introduced in commit https://github.com/sourcegraph/go-diff/commit/558247f853d5b5aa856bcab42659f40e895cae73#diff-b426b940498a3ecfc663fa2957b679f8R109, back when the `for {}` loop had a `break` in it.

Now, instead of that `break`, it performs this logic there instead. So removing this should be fine.